Jenkinsfile

```
@@ -1,13 +1,153 @@
+    }
+
     stages {
         stage('Initial setup') {
             steps {
-                sh 'echo Placeholder'
+                sh '''
+                    source "ci/functions.sh"
+                    set_label_flags
+
+                    > stage_flags
+                    echo "RELEASE_NAMESPACE:$RELEASE_NAMESPACE" >> stage_flags
+                    echo "SKIP_DEPLOY_EPHEMERAL:$SKIP_DEPLOY_EPHEMERAL" >> stage_flags
+                    echo "SKIP_IMAGE_BUILD:$SKIP_IMAGE_BUILD" >> stage_flags
+                    echo "SKIP_SMOKE_TEST:$SKIP_SMOKE_TEST" >> stage_flags
+                '''
+                script {
+                    FILE_CONTENTS = readFile('stage_flags')
+                    my_var_map = [:]
+                    my_vars = FILE_CONTENTS.split()
+                    for (i in my_vars) {
+                        s=i.split(':')
+                        if (s.length == 2) {
+                            my_var_map[s[0]] = s[1]
+                        } else {
+                            my_var_map[s[0]] = ""
+                        }
+                    }
+
+                    env.RELEASE_NAMESPACE = my_var_map['RELEASE_NAMESPACE']
+                    env.SKIP_BUILD = my_var_map['SKIP_BUILD']
+                    env.SKIP_IMAGE_BUILD = my_var_map['SKIP_IMAGE_BUILD']
+                    env.SKIP_DEPLOY_EPHEMERAL = my_var_map['SKIP_DEPLOY_EPHEMERAL']
+                    env.SKIP_SMOKE_TEST = my_var_map['SKIP_SMOKE_TEST']
+                }
+            }
+        }
+
+        stage('Build test image') {
+            when {
+                expression {
+                    return (! env.SKIP_IMAGE_BUILD)
+                }
+            }
+            steps {
+                script {
+                    withVault([configuration: configuration, vaultSecrets: secrets]) {
+                        sh '''
+                            source ./ci/functions.sh
+
+                            echo "Install bonfire repo/initialize, creating PR image"
+                            run_build_image_stage
+                        '''
+                    }
+                }
+            }
+        }
+
+        stage('Deploy Ephemeral Environment') {
+            when {
+                expression {
+                    return (! env.SKIP_DEPLOY_EPHEMERAL)
+                }
+            }
+            environment {
+                RELEASE_NAMESPACE="false"
+            }
+            steps {
+                script {
+                    withVault([configuration: configuration, vaultSecrets: secrets]) {
+                        NAMESPACE = sh(returnStdout: true, script: '''
+                            source ./ci/functions.sh
+                            _install_bonfire_tools
+
+                            source $CICD_ROOT/deploy_ephemeral_env.sh
+                        ''').trim()
+                        
+                        echo "Namespace reserved: ${NAMESPACE}"
```

Don't think this will  work as expected, one thing would be using files perhaps:

```suggestion
                          sh'''
                            source ./ci/functions.sh
                            _install_bonfire_tools
                            echo "$NAMESPACE" > .reserved_namespace

                            source $CICD_ROOT/deploy_ephemeral_env.sh
                        '''
                        NAMESPACE=sh(returnStdout: true, script: '''
                            cat .reserved_namespace
                         """
                        echo "Namespace reserved: ${NAMESPACE}"
```

While this may work it's not very nice - but it's due to the fact that the NAMESPACE is reserved (and exported to a variable) by the `deploy_ephemeral_env.sh`  script.